### PR TITLE
Add session refresh support to flask pyoidc

### PR DESCRIPTION
**Pull request by gdestuynder**
For feedback mainly, this probably need some pep8 and does need id token expiration time set as session time.

Feel free to merge if useful though.

Note this code also runs at https://python.testrp.security.allizom.org/

when it all looks pretty i'll PR back to upstream/original author

### CODE REVIEW

File: example/app.py [hunk 1]
```diff
@@ -6,10 +6,21 @@
...
 
-app.config.update({'SERVER_NAME': 'localhost:{}'.format(PORT),
-                   'SECRET_KEY': 'dev_key'})
-auth = OIDCAuthentication(app, issuer="https://localhost:50009")
-
+# See http://flask.pocoo.org/docs/0.12/config/
+app.config.update({'SERVER_NAME': 'example.com',
+                   'SECRET_KEY': 'dev_key',
+                   'PREFERRED_URL_SCHEME': 'https',
+                   'SESSION_PERMANENT': True, #turn on flask session support
```
Comment (gdestuynder) [inline example/app.py:13]: @andrewkrug any idea why this doesn't stick and the original author has to hardcode it?
Its also hard coded at #L128 now

**Comment by gdestuynder** [comment]
this has expiration support now

**Comment by darkwing** [comment]
Will there be a way to pull in this (and other work in this repo) via `pip` for `requirements.txt`?

**Comment by andrewkrug** [comment]
@darkwing We will either pull this into upstream or package our own artifact.  In a worse case you can reference a git repo in requirements.txt as well.

**Review by andrewkrug — APPROVED** [review]

**Comment by andrewkrug** [comment]
@gdestuynder +1 on this.  Let's merge it to master here so I can start pulling from master into the serverless-observatory.

**Comment by darkwing** [comment]
TIL I can pull a git repo into requirements.txt.  Thank you @andrewkrug

**Pull request merged**